test/jdk/java/awt/Clipboard/LostOwnershipChainTest/SystemClipboard2ProcTest.java

```diff
@@ -0,0 +1,170 @@
+*/
+
+import java.awt.Toolkit;
+import java.awt.datatransfer.Clipboard;
+import java.awt.datatransfer.ClipboardOwner;
+import java.awt.datatransfer.DataFlavor;
+import java.awt.datatransfer.StringSelection;
+import java.awt.datatransfer.Transferable;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+public class SystemClipboard2ProcTest {
+
+    public static void main(String[] args) throws Exception {
+        SystemClipboardOwner.run();
+
+        if (SystemClipboardOwner.failed) {
+            throw new RuntimeException("test failed: can not get actual " +
+                    "contents of the clipboard or set owner of the clipboard");
+        } else {
+            System.err.println("test passed");
+        }
+    }
+}
+
+class SystemClipboardOwner implements ClipboardOwner {
+    static volatile boolean failed;
+
+    private static final Object LOCK = new Object();
+
+    private static final int CHAIN_LENGTH = 5;
+    private final static Clipboard clipboard =
+        Toolkit.getDefaultToolkit().getSystemClipboard();
+
+    private int m, id;
+
+    public SystemClipboardOwner(int m) { this.m = m; id = m; }
+
+    public void lostOwnership(Clipboard cb, Transferable contents) {
+        System.err.println(id + " lost clipboard ownership");
+
+        Transferable t = getClipboardContents(cb, null);
+        // for test passing if t.getTransferData() will throw an exception
+        String msg = "" + (m + 1);
+        try {
+            msg = (String)t.getTransferData(DataFlavor.stringFlavor);
+        } catch (Exception e) {
+            System.err.println(id + " can't getTransferData: " + e);
+        }
+        System.err.println(id + " Clipboard.getContents(): " + msg);
+        if (!msg.equals("" + (m + 1))) {
+            failed = true;
+            System.err.println("Clipboard.getContents() returned incorrect contents!");
+        }
+
+        m += 2;
+        if (m <= CHAIN_LENGTH) {
+            System.err.println(id + " Clipboard.setContents(): " + m);
+            setClipboardContents(cb, new StringSelection(m + ""), this);
+        }
+        if (m >= CHAIN_LENGTH) {
+            synchronized (LOCK) {
+                LOCK.notifyAll();
+            }
+        }
+    }
+
+    public static void run() throws Exception {
+        SystemClipboardOwner cbo1 = new SystemClipboardOwner(0);
+        System.err.println(cbo1.m + " Clipboard.setContents(): " + cbo1.m);
+        setClipboardContents(clipboard, new StringSelection(cbo1.m + ""),
+                             cbo1);
+
+        ProcessBuilder pb = ProcessTools
+                .createTestJavaProcessBuilder(SystemClipboardOwner.class.getName());
+
+        Process process = ProcessTools.startProcess("Child", pb);
+        OutputAnalyzer outputAnalyzer = new OutputAnalyzer(process);
+
+        if (!process.waitFor(15, TimeUnit.SECONDS)) {
```

Comment:
In case of error it will be good to kill the child process via destroyForcibly(), or you can use
`OputAnalyzer executeTestJava(String... command)`
which will create the builder, then waits till completion and kill child process in case of errors.

Reply:
I prefer an early timeout, but `ProcessTools.executeProcess()` doesn't allow to set it, so `destroyForcibly()` was added.

Reply:
That looks fine. I guess people usually set that timeout via a jtreg parameter, and then executeProcess kills the child process when jtreg shuts down the test.